jina/types/arrays/document.py

```diff
@@ -534,9 +527,49 @@ def embeddings(self, emb: np.ndarray):
         :param emb: The embedding matrix to set
         """
 
-        assert len(emb) == len(
-            self
-        ), f'the number of rows in the input ({len(emb)}), should match the number of Documents ({len(self)})'
+        if len(emb) != len(self):
+            raise ValueError(
+                f'the number of rows in the input ({len(emb)}), should match the'
+                f'number of Documents ({len(self)})'
+            )
 
         for d, x in zip(self, emb):
             d.embedding = x
+
+    @property
+    def blobs(self) -> np.ndarray:
+        """Return a `np.ndarray` stacking all the `blob` attributes.
+
+        The `blob` attributes are stacked together along a newly created first
+        dimension (as if you would stack using ``np.stack(X, axis=0)``).
+
+        .. warning:: This operation assumes all blobs have the same shape and dtype.
+                 All dtype and shape values are assumed to be equal to the values of the
+                 first element in the DocumentArray / DocumentArrayMemmap
+
+        .. warning:: This operation currently does not support sparse arrays.
+
+        :return: blobs stacked per row as `np.ndarray`.
+        """
+        x_mat = b''.join(d.blob.dense.buffer for d in self._pb_body)
+        proto = self[0].proto.blob.dense
+
+        return np.frombuffer(x_mat, dtype=proto.dtype).reshape(
+            (len(self), *proto.shape)
+        )
+
+    @blobs.setter
+    def blobs(self, blobs: np.ndarray):
+        """Set the blobs of the Documents
+
+        :param emb: The blob array to set. The first axis is the "row" axis.
+        """
+
+        if len(blobs) != len(self):
+            raise ValueError(
+                f'the number of rows in the input ({len(blobs)}), should match the'
+                f'number of Documents ({len(self)})'
+            )
+
```

Comment:
this is replicated and the same code as in DA
maybe we should keep the setter in the parent class

Reply:
I guess so. Since the mixin does not seem to be an abstract class (does not raise `NotImplementedError`), this would be ok I think

Reply:
It turns out if subclass overrides the getter, it must also override the setter. For that reason this is not possible

Reply:
Found a way around it